src/twisted/runner/procmon.py

```diff
@@ -241,7 +246,8 @@ def startProcess(self, name):
 
         args, uid, gid, env = self.processes[name]
 
-        proto = LoggingProtocol()
+        if proto is None:
+            proto = LoggingProtocol()
         proto.service = self
         proto.name = name
```

Comment:
I wasn't sure what exactly to require in ILoggingProcessProtocol. I chose to require "service" and "name" attributes, but maybe specifying certain methods instead would make more sense. Then the previous four lines of code might need to be changed to something like:

```
if proto is None:
    proto = LoggingProtocol()
    proto.service = self
    proto.name = name
```

Please advise.